tron/serialize/runstate/statemanager.py

```diff
@@ -146,11 +150,21 @@ def restore(self, job_names, skip_validation=False):
         if not skip_validation:
             self._restore_metadata()
 
+        # First, restore the jobs themselves
         jobs = self._restore_dicts(runstate.JOB_STATE, job_names)
         # jobs should be a dictionary that contains  job name and number of runs
         # {'MASTER.k8s': {'run_nums':[0], 'enabled': True}, 'MASTER.cits_test_frequent_1': {'run_nums': [1,0], 'enabled': True}}
-        for job_name, job_state in jobs.items():
-            job_state["runs"] = self._restore_runs_for_job(job_name, job_state)
+
+        # second, restore the runs for each of the jobs restored above
+        with concurrent.futures.ThreadPoolExecutor(max_workers=3) as executor:
+            # start the threads and mark each future with it's job name
+            # this is useful so that we can index the job name later to add the runs to the jobs dictionary
+            results = {
+                executor.submit(self._restore_runs_for_job, job_name, job_state): job_name
+                for job_name, job_state in jobs.items()
+            }
```

Comment:
question: should this be a list of futures? i'm kinda surprised that this works - i would have expected `concurrent.futures.as_completed` to be unhappy with getting passed a dict

Reply:
yup, it works i got the inspiration from this example: https://docs.python.org/3/library/concurrent.futures.html#threadpoolexecutor-example . Basically I mark each future with its job name because it makes it easy in the code below to index the map and get the "job_name" which is later used to index "jobs" and save under each job name ["runs"], the result of this func "_restore_runs_for_job" 
```
jobs[results[result]]["runs"] = result.result()
```